VenafiPS/Public/Test-TppToken.ps1

```diff
@@ -0,0 +1,170 @@
+                    $true
+                } else {
+                    throw 'Please enter a valid server, https://venafi.company.com or venafi.company.com'
+                }
+            }
+        )]
+        [Alias('Server')]
+        [string] $AuthServer,
+
+        [Parameter(Mandatory, ParameterSetName = 'AccessToken', ValueFromPipeline)]
+        [PSCredential] $AccessToken,
+
+        [Parameter(Mandatory, ParameterSetName = 'TppToken', ValueFromPipeline)]
+        [pscustomobject] $TppToken,
+
+        [Parameter()]
+        [switch] $GrantDetail,
+
+        [Parameter(ParameterSetName = 'Session')]
+        [VenafiSession] $VenafiSession = $script:VenafiSession
+    )
+
+    begin {
+        $params = @{
+            Method  = 'Get'
+            UriRoot = 'vedauth'
+            UriLeaf = 'Authorize/Verify'
+        }
+    }
+
+    process {
+        
+        Write-Verbose ('Parameter set: {0}' -f $PSCmdlet.ParameterSetName)
+
+        switch ($PsCmdlet.ParameterSetName) {
+            'Session' {
+                $VenafiSession.Validate('token') | Out-Null
+                $params.VenafiSession = $VenafiSession
+            }
+
+            'AccessToken' {
+                $AuthUrl = $AuthServer
+            # add prefix if just server url was provided
+                if ( $AuthServer -notlike 'https://*') {
+                    $AuthUrl = 'https://{0}' -f $AuthUrl
+                }
+
+                $params.ServerUrl = $AuthUrl
+                $params.Header = @{'Authorization' = 'Bearer {0}' -f $AccessToken.GetNetworkCredential().password }
+            }
+
+            'TppToken' {
+                if ( -not $TppToken.Server -or -not $TppToken.AccessToken ) {
+                    throw 'Not a valid TppToken'
+                }
+
+                $params.ServerUrl = $TppToken.Server
+                $params.Header = @{'Authorization' = 'Bearer {0}' -f $TppToken.AccessToken.GetNetworkCredential().password }
+            }
+
+            Default {
+                throw ('Unknown parameter set {0}' -f $PSCmdlet.ParameterSetName)
+            }
+        }
+
+        Write-Verbose ($params | Out-String)
+
+        if ($GrantDetail) {
+            $response = Invoke-VenafiRestMethod @params
+
+            [PSCustomObject] @{
+                Application    = $response.application
+                AccessIssued   = ([datetime] '1970-01-01 00:00:00').AddSeconds($response.access_issued_on_unix_time)
+                GrantIssued    = ([datetime] '1970-01-01 00:00:00').AddSeconds($response.grant_issued_on_unix_time)
+                Scope          = $response.scope
+                Identity       = $response.identity
+                RefreshExpires = ([datetime] '1970-01-01 00:00:00').AddSeconds($response.expires_unix_time)
+                ValidFor       = $response.valid_for
+            }
+
+        } else {
+            $response = Invoke-VenafiRestMethod @params -FullResponse
+
+            switch ([int]$response.StatusCode) {
```

Comment:
do you only want to check status code when GrantDetail isn't provided?  I would think you'd ensure you get a 200 and then build the response?

Reply:
added logic to check for 200 to this switch as well